backend/test_observer/controllers/test_executions/logic.py

```diff
@@ -13,7 +13,23 @@
 from test_observer.data_access.models_enums import TestExecutionStatus, TestResultStatus
 from test_observer.data_access.repository import get_or_create
 
-from .models import C3TestResult, C3TestResultStatus
+from .models import C3TestResult, C3TestResultStatus, StartTestExecutionRequest
+
+
+def reset_test_execution(
+    request: StartTestExecutionRequest,
+    db: Session,
+    test_execution: TestExecution,
+):
+    test_execution.status = TestExecutionStatus.IN_PROGRESS
+    test_execution.ci_link = request.ci_link
+    test_execution.c3_link = None
+    test_execution.review_decision = []
+    test_execution.review_comment = ""
+    db.execute(
+        delete(TestResult).where(TestResult.test_execution_id == test_execution.id)
+    )
```

Comment:
Should we use the `delete_previous_results` function here, to make sure we only maintain one place where we delete TestResults.